patches/2025-02-20-pagination-cursor-field.lua

```diff
@@ -0,0 +1,96 @@
+--[[
+    Fixes a bug in pagination where a unique cursor field was not being used for pagination, causing non-deterministic results for paginated requests.
+    By default, all tables will be sorted using the unique cursor field, and then sorted by the requested sortBy field.
+    This ensures that the same cursor will always return the same results.
+
+    Reviewers: Dylan, Jonathon, Phil, Ariel
+]]
+--
+
+local utils = require(".src.utils")
+
+_G.package.loaded[".src.utils"].paginateTableWithCursor = function(
+	tableArray,
+	cursor,
+	cursorField,
+	limit,
+	sortBy,
+	sortOrder
+)
+	-- Sort first by cursorField for a stable sort
+	if cursorField ~= nil then
+		table.sort(tableArray, function(a, b)
+			return a[cursorField] < b[cursorField]
+		end)
+	end
+
+	local sortedArray = utils.sortTableByFields(tableArray, { { order = sortOrder, field = sortBy } })
+
+	if not sortedArray or #sortedArray == 0 then
+		return {
+			items = {},
+			limit = limit,
+			totalItems = 0,
+			sortBy = sortBy,
+			sortOrder = sortOrder,
+			nextCursor = nil,
+			hasMore = false,
+		}
+	end
+
+	local startIndex = 1
+
+	if cursor then
+		for i, obj in ipairs(sortedArray) do
+			if cursorField and obj[cursorField] == cursor or cursor == obj then
```

Comment:
Hmm reviewing this again I now worry about what happens if obj is not indexable as a table and we attempt to index it by the cursorField 🤔